src/clr-angular/forms/checkbox/checkbox-wrapper.ts

```diff
@@ -4,34 +4,58 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { OnInit, Component, ContentChild } from '@angular/core';
+import { Component, OnInit, ContentChild, Inject, InjectionToken, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Subscription } from 'rxjs';
 
 import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
 import { ControlIdService } from '../common/providers/control-id.service';
 import { ClrLabel } from '../common/label';
 
+/* tslint:disable-next-line:variable-name */
+export const TogglerService = new InjectionToken<any>(undefined);
```

Comment:
Isn't it cleaner to have a designated service class instead of using `InjectionToken`?

Reply:
Actually this is fine, but can you use `InjectionToken<Subject<boolean>` instead, so we get proper type checking? Thanks.

I also just realized I missed the same comment on the `ToggleService` for the password input. If you want to squeeze it in the PR (even though it has nothing to do with it) you can, otherwise we'll just fix it with a one-line PR later.

Also, you wouldn't have to disable TSLint if we matched this to our conventions. We do this in a couple places, and we use actual constant names since we're not truly defining a service but just an injection token and a provider. In this case, you'd have:
```typescript
export const IS_TOGGLE = new InjectionToken<Subject<boolean>>("IS_TOGGLE");
export function isToggleFactory() {
  return new BehaviorSubject<boolean>(false);
}
export const IS_TOGGLE_PROVIDER = { provide: IS_TOGGLE, useFactory: isToggleFactory };
```
and then your component simply declares: `providers: [ControlIdService, IS_TOGGLE_PROVIDER]`.
Note that the string argument in the `InjectionToken` constructor is here purely for error messages and debugging purposes, so it's good practice to always set it. It's not public anyway.

Reply:
Updated.